Description: Port arptbl_tuneup to Debian
 Adjust network configuration path and utility paths in
 arptbl_tuneup so that it will work correctly on Debian.
Author: Brian T. Smith <bsmith@systemfabricworks.com>
Forwarded: not-needed
Last-Update: <2017-11-20>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/arptbl_tuneup
+++ b/arptbl_tuneup
@@ -52,10 +52,16 @@
 # limit table size
 max_entries=65536
 
+NETWORK_CONFIG=
+DEBIAN=0
+
 if [[ -d /etc/sysconfig/network-scripts ]]; then
-       NETWORK_CONFIG="/etc/sysconfig/network-scripts"
+	NETWORK_CONFIG="/etc/sysconfig/network-scripts"
 elif [[ -d /etc/sysconfig/network ]]; then
-       NETWORK_CONFIG="/etc/sysconfig/network"
+	NETWORK_CONFIG="/etc/sysconfig/network"
+elif [[ -f /etc/network/interfaces ]]; then
+	NETWORK_CONFIG="/etc/network/interfaces"
+	DEBIAN=1
 fi
 
 ###########################################################
@@ -109,6 +115,15 @@
 	return $ret
 }
 
+get_netmask_cidr_debian()
+{
+	local iface=$1
+	local inet=$2
+	ret=$(get_iface_netmask_cidr $iface $inet)
+	return $ret;
+}
+
+
 #################################################
 # Get CIDR netmask value from running interface
 #
@@ -270,8 +285,11 @@
 		_num_nodes=0
 		for dev in $devlist; do
 			# echo -n "lookup $dev for $inet netmask..."
-
-			get_netmask_cidr $dev $inet
+			if [ $DEBIAN -eq 1 ] ; then
+				get_netmask_cidr_debian $dev $inet
+		    else
+   				get_netmask_cidr $dev $inet
+			fi
 			_ret=$?
 			
 			if ((_ret < 2 )); then
@@ -297,7 +315,7 @@
 			((cidrmask = addrsize - max_maskwidth))
         	fi
 
-		# echo "found $_found interface(s) for $inet, max netmask: /$cidrmask"
+			#echo "found $_found interface(s) for $inet, max netmask: /$cidrmask"
 
 	        ((_num_nodes = 1 << (addrsize - cidrmask)))
 
@@ -366,7 +384,7 @@
 Usage()
 {
 	echo >&2
-	echo "Usage: `/bin/basename $0` (--help|start|stop|restart|force-reload|status)" >&2
+	echo "Usage: `/usr/bin/basename $0` (--help|start|stop|restart|force-reload|status)" >&2
 	echo "    --help - produce full help text" >&2
 	echo >&2
 	echo "    Must execute as root. Adjusts kernel arp/neighbor table sizes for very large" >&2
@@ -378,7 +396,7 @@
 Usage_full()
 {
 	echo >&2
-	echo "Usage: `/bin/basename $0` (--help|start|stop|restart|force-reload|status)" >&2
+	echo "Usage: `/usr/bin/basename $0` (--help|start|stop|restart|force-reload|status)" >&2
 	echo "    --help - produce full help text" >&2
 	echo >&2
 	echo "    Must execute as root. Adjusts kernel arp/neighbor table sizes for very large" >&2
